test/lib/utils/decoder_test.rb

```diff
@@ -0,0 +1,74 @@
+require "test_helper"
+require "jwt"
+require "openssl"
+require "json"
+require "tempfile"
+require_relative "../../../app/lib/utils/decoder_factory"
+require_relative "../../../app/lib/utils/jwks_decoder"
+
+class DecoderTest < ActiveSupport::TestCase
+  test "jwksdecoder.decode returns correct identity" do
+    jwk = generate_jwks_signing_key
+    token = generate_jwks_token(jwk)
+    keyset_path = generate_jwks_keyset(jwk)
+
+    identity = JwksDecoder.new(keyset_path).decode(token)
```

Comment:
i'd suggest a separate test file for each of the decoders here, even though they would only have one test each

Reply:
done